src/app/static/src/js/components/runWorkflow/runWorkflowReport.vue

```diff
@@ -444,8 +449,16 @@ export default Vue.extend<Data, Methods, Computed, Props>({
         stepIsValid(newVal) {
             this.$emit("valid", newVal);
         },
-        reportsOrigin(newVal) {
+        reportsOrigin: async function (newVal) {
             session.setSelectedWorkflowReportSource(newVal);
+
+            const newlyActiveReports = this.inactiveOriginWorkflowReports;
+            this.inactiveOriginWorkflowReports = [...this.workflowMetadata.reports];
+            await this.updateWorkflowReports(newlyActiveReports);
+            session.setInactiveOriginWorkflowReports(this.inactiveOriginWorkflowReports);
+
+            // Revalidate reports as branch may have changed while other origin was active
+            this.validateWorkflowReportsWithGit()
```

Comment:
```suggestion
            await this.validateWorkflowReportsWithGit()
```

Reply:
The watcher doesn't actually need an await here - it doesn't need to use the result of `validateWorkflowReportsWithGit` which can complete and have any effects in its own time.